htlcswitch/hop/payload.go

```diff
@@ -180,6 +186,7 @@ func NewPayloadFromReader(r io.Reader) (*Payload, error) {
 		},
 		MPP:           mpp,
 		AMP:           amp,
+		metadata:      metadata,
```

Comment:
Here we'll have a non-nil, but empty `metadata` field if there was no value read. Should we check `parsedTypes` like we do for `MPPOnionType`/`AMPOnionType`?

Reply:
Indeed, if we want to distinguish between zero length metadata and no metadata, I need to change this.

Reply:
Changed